File: leetsync/2394-count-subarrays-with-score-less-than-k/count-subarrays-with-score-less-than-k.go
func countSubarrays(nums []int, k int64) int64 {
	n := len(nums)
	result, sum := 0, 0
	left, right := 0, 0
	for right < n {
		sum += nums[right]
		for left <= right && (sum*(right-left+1) >= int(k)) {
			sum -= nums[left]
			left++
		}
		result += (right - left + 1)
		right++
	}
	return int64(result)
}

File: leetsync/2394-count-subarrays-with-score-less-than-k/count-subarrays-with-score-less-than-k.java
class Solution {
    public long countSubarrays(int[] nums, long k) {
        int n = nums.length;
        long result = 0, sum = 0;
        int left = 0, right = 0;
        while (right < n) {
            sum += nums[right];
            while (left <= right && (sum * (right - left + 1) >= k)) {
                sum -= nums[left];
                left++;
            }
            result += (right - left + 1);
            right++;
        }
        return result;
    }
}